### ISSUE TITLE
Does this mean the volume is corrupted?

### ISSUE THREAD

**Issue by nicgentile** [mhogomchungu/zuluCrypt#67]
Hi,

I've been getting an error on my volumes after trying to mount them using
zuluCrypt-cli -o -d /dev/sdX1 -m sdX1 -e rw -p after a power blackout.

"ERROR: Failed to mount a filesystem:invalid/unsupported mount option or unsupported file system encountered"

Where I am, this is quite common, and unfortunately, majority of the times the mitigation is hardly effective. Is there any way to come back from this?

Also, what exactly do I need to type into the computer to export headers? I've struggled with this for nearly two days, so be as verbose as possible.

Thanks.

**Comment by nicgentile** [comment]
Hi,

I've been getting an error on my volumes after trying to mount them using
`zuluCrypt-cli -o -d /dev/sdX1 -m sdX1 -e rw -p ` after a power blackout. 

"ERROR: Failed to mount a filesystem:invalid/unsupported mount option or unsupported file system encountered"

Where I am, this is quite common, and unfortunately, majority of the times the mitigation is hardly effective. Is there any way to come back from this? 

Also, what exactly do I need to type into the computer to export headers? I've struggled with this for nearly two days, so be as verbose as possible.

Thanks.

Nicky

**Comment by mhogomchungu** [comment]
The file system on the volume is corrupted.

To unlock the volume without mounting it,log in to root's account and then use below command:

```
zuluCrypt-cli -O -d /dev/sdX1
```

Try to repair the file system using ```fsck``` tool on the encrypted mapper device located at 
```/dev/mapper/zuluCrypt-0-NAAN-sdX1-XXYZ```

Read up on fsck options to see how to use it.

**Comment by nicgentile** [comment]
Hi,

I know how to use fsck, but had no idea you could do all that. Went ahead and re-encrypted the corrupted drives, so I can only try this out if it happens again. Nothing valuable was lost, and I am still learning, so nothing to worry about.

**Comment by mhogomchungu** [comment]
A command to run to export a LUKS header on a device is:
```
zuluCrypt-cli -B -d /dev/sdX1 -z luks -f /path/to/where/to/store/the/header
```

A command to run to restore a LUKS header from a backup is:
```
zuluCrypt-cli -R -d /dev/sdX1 -z luks -f /path/to/where/to/store/the/header
```

**Comment by nicgentile** [comment]
Hi,

We had a blackout so it went down. When it came back I tried 

`zuluCrypt-cli -O -d /dev/sdX1`

which told me

`SUCCESS: luks volume opened successfully`

Then I tried fsck to which I got

```
fsck.ext4: No such file or directory while trying to open /dev/mapper/zuluCrypt-0-NAAN-sdb1-XXYZ
Possibly non-existent device?
```
I run a df -h check and the mapper does not appear. I tried with a non-corrupt volume as well and I get the same problem. Mounts successfully but mapper does not appear.

What am I missing?
